### ISSUE TITLE
Fatal error: Uncaught TypeError: date() expects parameter 2 to be integer, string given

### ISSUE THREAD

**Issue by Veraxus** [sebastianbergmann/php-code-coverage#708]
| Q                         | A
| --------------------------| ---------------
| php-code-coverage version | 7.0.8
| PHP version               | 7.2.24
| Driver                    | ??
| Xdebug version (if used)  | ??
| Installation Method       | Composer
| Usage Method              | Codeception + PhpUnit
| PHPUnit version (if used) | 8.4.3

I recently updated from PhpUnit 7.5.13 + Code Coverage and the CI job started self-destructing as a result of a line in php-code-coverage that references `$_SERVER['REQUEST_TIME']`

On investigation, the value of that variable is `"1573623209"`

This is running inside the official php Docker image for php 7.2-apache

```
Fatal error: Uncaught TypeError: date() expects parameter 2 to be integer, string given in /vendor/phpunit/php-code-coverage/src/Report/Text.php on line 157

TypeError: date() expects parameter 2 to be integer, string given in /vendor/phpunit/php-code-coverage/src/Report/Text.php on line 157

Call Stack:
    0.0006     454800   1. {main}() /vendor/codeception/codeception/codecept:0
    0.2081    6463928   2. Codeception\Application->run() /vendor/codeception/codeception/codecept:43
    0.2081    6463928   3. Codeception\Application->run() /vendor/codeception/codeception/src/Codeception/Application.php:107
    0.2218    6654608   4. Codeception\Application->doRun() /vendor/symfony/console/Application.php:149
    0.2254    6667240   5. Codeception\Application->doRunCommand() /vendor/symfony/console/Application.php:273
    0.2254    6667240   6. Codeception\Command\Run->run() /vendor/symfony/console/Application.php:934
    0.2296    6671112   7. Codeception\Command\Run->execute() /vendor/symfony/console/Command/Command.php:255
    3.6694   35383488   8. Codeception\Codecept->printResult() /vendor/codeception/codeception/src/Codeception/Command/Run.php:412
    3.6704   35386248   9. Symfony\Component\EventDispatcher\EventDispatcher->dispatch() /vendor/codeception/codeception/src/Codeception/Codecept.php:210
    3.6706   35388704  10. Symfony\Component\EventDispatcher\EventDispatcher->callListeners() /vendor/symfony/event-dispatcher/EventDispatcher.php:73
    3.6706   35388704  11. Symfony\Component\EventDispatcher\EventDispatcher->doDispatch() /vendor/symfony/event-dispatcher/EventDispatcher.php:235
    3.6707   35388704  12. Codeception\Coverage\Subscriber\Printer->printResult() /vendor/symfony/event-dispatcher/EventDispatcher.php:260
    3.6707   35388704  13. Codeception\Coverage\Subscriber\Printer->printConsole() /vendor/codeception/codeception/src/Codeception/Coverage/Subscriber/Printer.php:63
    3.6713   35486608  14. SebastianBergmann\CodeCoverage\Report\Text->process() /vendor/codeception/codeception/src/Codeception/Coverage/Subscriber/Printer.php:98
    6.3178   38589368  15. date() /vendor/phpunit/php-code-coverage/src/Report/Text.php:157




FATAL ERROR. TESTS NOT FINISHED.
Uncaught TypeError: date() expects parameter 2 to be integer, string given in /vendor/phpunit/php-code-coverage/src/Report/Text.php:157
Stack trace:
#0 /vendor/phpunit/php-code-coverage/src/Report/Text.php(157): date('  Y-m-d H:i:s', '1573620950')
#1 /vendor/codeception/codeception/src/Codeception/Coverage/Subscriber/Printer.php(98): SebastianBergmann\CodeCoverage\Report\Text->process(Object(SebastianBergmann\CodeCoverage\CodeCoverage), false)
#2 /vendor/codeception/codeception/src/Codeception/Coverage/Subscriber/Printer.php(63): Codeception\Coverage\Subscriber\Printer->printConsole(Object(Codeception\PHPUnit\ResultPrinter\UI))
#3 /vendor/symfony/event-dispatcher/EventDispatcher.php(260): Codeception\Coverage\Subscriber\Printer->printResult(Object(Codeception\Event\PrintResultEvent), 'result.print.af...', Ob 
in /vendor/phpunit/php-code-coverage/src/Report/Text.php:157
```

**Comment by sebastianbergmann** [comment]
The type of `$_SERVER['REQUEST_TIME']` is `int`, as can be seen [here](https://3v4l.org/6vffR).

I could, of course, cast `$_SERVER['REQUEST_TIME']` to `int` but that would only hide a problem that should not exist. Something in your environment overwrites `$_SERVER['REQUEST_TIME']` with a value of type `string`. This is wrong and needs to be fixed.

**Comment by Veraxus** [comment]
It doesn't appear to be environmental. The recast from integer to string happens somewhere between these steps:
```
    0.5124    7126176   7. Codeception\Command\Run->execute() /var/www/html/vendor/symfony/console/Command/Command.php:255
    4.1690   37504776   8. Codeception\Codecept->printResult() /var/www/html/vendor/codeception/codeception/src/Codeception/Command/Run.php:412
```
It's still an integer at step 7, but by step 8 it's a string. I don't see any explicit recasting going on between those steps, though. Possibly something in Symfony? *sigh*

**Comment by sebastianbergmann** [comment]
So either Codeception or Symfony is doing something to `$_SERVER['REQUEST_TIME']`.

**Comment by Veraxus** [comment]
Found it. It's WordPress core's `wp_magic_quotes()` functionality. It's been there since 2010 so there has to have been a change somewhere else causing this to break. Unless you can explicitly cast the variable as int on the date format call, it looks like code coverage won't work with WordPress any more.

**Comment by sebastianbergmann** [comment]
Sorry, but that is an issue in Wordpress that needs to be fixed in Wordpress. I do not like working around other software's issues in my software.

**Comment by Veraxus** [comment]
As someone who has, in the past, contributed to WordPress core - I can confidently say that given the climate in that community: that will never, ever happen. WordPress core is bursting at the seams with stupidity and tech debt that will never be fixed - not because there aren't people that want to, but because there's such strong resistance to it from the project's maintainers. Such a shame :-(

So for posterity - for now, I'm working around this by adding the following to my `tearDown()` methods... 
```
$_SERVER['REQUEST_TIME'] = (int) $_SERVER['REQUEST_TIME'];
```

**Comment by sebastianbergmann** [comment]
CC @jrfnl

**Comment by jrfnl** [comment]
I noticed @Veraxus had already opened [an issue in WP Trac](https://core.trac.wordpress.org/ticket/48605).

I concur that this should be changed in WP and have just uploaded two patches, either of which will fix the immediate problem.

I can't make any promises, but have added a comment arguing the case to get this committed. :crossed_fingers:

**Comment by jrfnl** [comment]
> As someone who has, in the past, contributed to WordPress core - I can confidently say that given the climate in that community: that will never, ever happen. WordPress core is bursting at the seams with stupidity and tech debt that will never be fixed

Just in case anyone wants to know the status:
* A basic patch which fixes this issue was included in WordPress 5.4 which was released on March 31 2020.
* A more structural patch which fixes the underlying logic stupidity (yes, I concur on that) is still being discussed, but will hopefully go into WordPress 5.5.

So, I'm happy to say that code coverage should work again in a WP environment and that we _can_ get things fixed in WP. Please let the record reflect that :judge:

**Comment by Veraxus** [comment]
@jrfnl This warms the ice-cold cockles of my heart. Thank you!
